### ISSUE TITLE
Allow Fitting of Models with All Inputs Fixed

### ISSUE THREAD

**Issue by DanRyanIrish** [astropy/astropy#17550]
### What is the problem this feature will solve?

There are uses cases where the input grids of a model are all fixed using the `fixed_inputs` function.  However, such models still need to be fit to data as their parameters are not fixed.  Yet, when all inputs are fixed in an astropy, the model becomes not `Fittable`.

For example, we need this use case to work:
```python
import numpy as np
from astropy.modeling import Fittable1DModel, Parameter, fitting
from astropy.modeling.models import Mapping, fix_inputs
from astropy.modeling.functional_models import Linear1D

photon = Linear1D()  # model with one input, x.
p = fix_inputs(photon, {'x': np.array([1])})  # Fix the x input.
fitter = fitting.LevMarLSQFitter()  
res = fitter(p, y=np.array([1]))  # Fit data with fixed/no inputs.
# The above line raises an error.
```

### Describe the desired outcome

To solve this, we request that astropy fitting support fitting models with all inputs fixed, or equivalently, no inputs.

**Comment by Cadair** [comment]
From a quick poke about this there are two issues:

* If you use `fix_inputs` then the resulting model does not have `fittable = True`, not entirely sure why.
* If you have a model with no inputs the API for `__call__` on a fitter doesn't let you not pass `x=`.

**Comment by dsshao821** [comment]
Looked into this, it seems there was an issue in the `_make_leaflist` code that didn't like the model that `fix_inputs` created, specifically that the right-hand side wasn't a submodel. I edited this in the PR above.

I am not sure it makes sense to change the code to allow for a model with no inputs, so I left the second issue unaddressed.

**Comment by Cadair** [comment]
yeah, that didn't work. I don't think it propagated correctly iirc.
